lib/collection/src/collection_manager/segments_updater.rs

```diff
@@ -51,6 +51,8 @@ pub(crate) fn delete_points(
             batch,
             |_| (),
             |id, _idx, write_segment, ()| write_segment.delete_point(op_num, id, hw_counter),
+            // Apply point delete to all point versions
+            true,
```

Comment:
Critical bit: for point delete operations we use `true` to apply the delete to all point versions.

For other operations this flag is false, meaning it only applies to the latest version as expected.